src/PostgREST/Auth.hs

```diff
@@ -47,15 +48,15 @@ jwtClaims _ "" "" = return $ JWTClaims M.empty
 jwtClaims secret audience payload =
   case secret of
     Nothing -> return JWTMissingSecret
-    Just jwk -> do
-      let validation = set audiencePredicate (==  fromString audience) defaultJWTValidationSettings
+    Just k -> do
+      let audience' = fromJust $ preview stringOrUri audience
```

Comment:
Huh, `preview` is cool, I never knew about that.

However `fromJust` is an unsafe function that dies on Nothing. Is it possible to make this code more robust?

Reply:
I hope there is one ... It comes directly from the `jose` [example](https://github.com/frasertweedale/hs-jose/blob/master/example/Main.hs#L82). I also find this line quite unsatisfying.

Reply:
Just asked here: https://github.com/frasertweedale/hs-jose/issues/58

Reply:
If you're dealing with a user-supplied value you should treat the `Nothing` case a user error.
If it is a "known good" value then use `fromJust . preview stringOrUri` without remorse.

Reply:
@begriffs Do you wish to consider the Nothing case as a configuration error or leave it as before this PR. I would argue this is good to merge because it does not modify the old behaviour (but I am not convinced the old behaviour was sane)

Reply:
OK, I figured out how to trigger this edge case. Use a colon in the desired audience, but make sure it's not part of a valid url:

```ini
jwt-aud = "not::a url ? at all"
```

Then make a request with a valid jwt that contains an aud claim. It triggers an internal server error.

On the other hand I tried postgrest on the master branch in this same situation and it responds with HTTP 401:

```json
{
  "message": "JWTNotInAudience"
}
```

I don't know why the new jose library insists on parsing correct urls in the presence of a colon. Must be a more correct interpretation of the spec or something.

Given that the library is acting this way, it's probably best if we change the type of `configJwtAudience` to `StringOrURI`. https://github.com/begriffs/postgrest/blob/master/src/PostgREST/Config.hs#L62

The parsing would need to change here as well - https://github.com/begriffs/postgrest/blob/master/src/PostgREST/Config.hs#L121

Reply:
From https://tools.ietf.org/html/rfc7519#page-5:

 >     StringOrURI
 >        A JSON string value, with the additional requirement that while
 >        arbitrary string values MAY be used, any value containing a ":"
 >        character MUST be a URI [RFC3986].  StringOrURI values are
 >        compared as case-sensitive strings with no transformations or
 >        canonicalizations applied.

You can use the `stringOrUri` prism for parsing a `StringOrURI` value.

Reply:
Cool. I do like the correctness of the library, it forces us to do things right, like having a sufficient key length, or now this audience parsing.